src/Microsoft.ML.Tokenizers/Microsoft.ML.Tokenizers.csproj

```diff
@@ -21,4 +21,93 @@
     <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonVersion)" />
   </ItemGroup>
 
+  <UsingTask TaskName="CompressFile"
+    TaskFactory="RoslynCodeTaskFactory"
+    AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
+    <ParameterGroup>
+      <Files ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System.IO" />
+      <Using Namespace="System.IO.Compression" />
+      <Code Type="Fragment" Language="cs">
+			<![CDATA[
+		    byte[] newLine = new byte[] { (byte)'\n' };
+
+        foreach(var file in Files)
+        {
+            using var sourceStream = File.OpenRead(file.GetMetadata("FullPath"));
+
+            string line;
+            int lineNumber = 0;
+            using MemoryStream ms = new MemoryStream();
+            using StreamReader reader = new StreamReader(sourceStream);
+
+            while ((line = reader.ReadLine()) != null)
+            {
+                if (line.Length == 0) { continue; }
```

Comment:
Why do we skip empty lines without incrementing lineNumber?

Reply:
I see I guess if they were relevant to token ID we would insert them later on. Can you rename lineNumber to destLineNumber?